src/Core/Application/Security/UseCase/RenewPassword/RenewPassword.php

```diff
@@ -49,18 +50,26 @@ public function __invoke(
         RenewPasswordPresenterInterface $presenter,
         RenewPasswordRequest $renewPasswordRequest
     ): void {
+        //Get User informations
         $user = $this->readRepository->findUserByAlias($renewPasswordRequest->userAlias);
         if ($user === null) {
             $presenter->setResponseStatus(new NotFoundResponse('User'));
             return;
         }
-        if (password_verify($renewPasswordRequest->oldPassword, $user->getPassword()) === false) {
+
+        //Validate that old password matches the current user password
+        if (password_verify($renewPasswordRequest->oldPassword, $user->getPassword()->getPasswordValue()) === false) {
             $presenter->setResponseStatus(new ErrorResponse('Invalid credentials'));
             return;
         }
-        //@Todo: Check that new password follow the security policy.
 
-        $newPassword = password_hash($renewPasswordRequest->newPassword, \CentreonAuth::PASSWORD_HASH_ALGORITHM);
+        //@todo: Check that new password follow the security policy.
+
+        //@todo: If use ancient password is not allowed, check that new password wasn't already used.
+
+        //Create the new password and set it to the user.
+        $newPasswordValue = password_hash($renewPasswordRequest->newPassword, \CentreonAuth::PASSWORD_HASH_ALGORITHM);
```

Comment:
I would have preferred a call to an external method because it is not the role of the UseCase to create the password.
Furthermore, this would allow us to always call the same method for password creation, regardless of how the password is generated

Reply:
Agreed, if you are ok with that, I'll handle that in the PR for Implementation of the UseCase(with all the @todo remaining)

Reply:
ok